server/app/controllers/applicant/ApplicantProgramBlocksController.java

```diff
@@ -376,27 +376,62 @@ public CompletionStage<Result> confirmAddressWithApplicantId(
         applicantRequestedActionWrapper.getAction());
   }
 
-  /** Handles the applicant's selection from the address correction options. */
+  /**
+   * Handles the applicant's selection from the address correction options.
+   *
+   * @param request the HTTP request containing context and session information
+   * @param programParam the program identifier, which can be either a program slug or program ID,
+   *     depending on feature configuration
+   * @param blockId the unique identifier of the specific block within the program containing the
+   *     address question
+   * @param inReview indicates whether the applicant is currently in review mode (true) or edit mode
+   *     (false), which affects navigation behavior after address confirmation
+   * @param applicantRequestedActionWrapper wrapper containing the applicant's address selection
+   *     choice and related context for processing the address confirmation
+   * @param isFromUrlCall indicates whether this method was invoked directly from a URL call, used
+   *     to determine redirect behavior when program slug URLs are enabled
+   * @return a CompletionStage that resolves to a Result handled by confirmAddressWithApplicantId()
+   */
   @Secure
   public CompletionStage<Result> confirmAddress(
       Request request,
-      long programId,
+      String programParam,
       String blockId,
       boolean inReview,
-      ApplicantRequestedActionWrapper applicantRequestedActionWrapper) {
-    Optional<Long> applicantId = getApplicantId(request);
-    if (applicantId.isEmpty()) {
+      ApplicantRequestedActionWrapper applicantRequestedActionWrapper,
+      Boolean isFromUrlCall) {
```

Comment:
question (non-blocking): can you help me understand why we need this `isFromUrlCall` check? why should it be handled differently if we go through the `confirmAddress` method in `ApplicantRoutes`?

Reply:
thinking about this more -- is there another way we get to this method without going through applicant routes? since this is the address suggestions page, i think it would impossible for someone to navigate here directly via url.

Reply:
I think see your point. We use `isFromUrl` on the GET calls to redirect direct URL calls using program id when the program slug is enabled. Since it's a POST call, there is no direct URL call and there is no need to handle that case.

Taking a step back, since the POST route is only used through routes, then the route will always receive the program id. Thus, if the goal is to change all 'external' URLs to use program id then we don't need to update the POST routes and can leave them using program id. WDYT?

Reply:
that makes sense to me!